Dev.jmmartinezv@ucenfotec.ac.cr/vbs22 110 configuraci n para el consumo de una api de ia

mur-tz opened the pull request.

mur-tz committed:
Implement OpenAIService for generating travel suggestions

- Add OpenAIService class with method to generate travel suggestions based on user query and system prompt
- Use Jackson ObjectMapper for JSON serialization
- Configure RestTemplate for HTTP requests
- Include necessary Spring annotations and dependency injections
- 1000 token limit per request. Might be changed though

mur-tz committed:
Refactor CountryController and CurrencyController

- Add missing @RequestMapping annotations for endpoints
- Ensure proper HTTP method mappings for CRUD operations
- Improve consistency and readability of controller methods
- Enhance RESTful API design adherence

mur-tz committed:
feat: add integration test for OpenAIService

- Created `OpenAIServiceTestConfig` to isolate test environment
- Added necessary beans in `OpenAIServiceTestConfig`
- Updated `OpenAIServiceIntegrationTests` to use the new test configuration
- Implemented test for `generateTravelSuggestions` method in `OpenAIService`

mur-tz committed:
Comment Integration test

mur-tz commented on demo/src/main/java/com/project/demo/logic/request/OpenAIService.java:
```diff
@@ -0,0 +1,69 @@
+    public OpenAIService(RestTemplate restTemplate, ObjectMapper objectMapper) {
+        this.restTemplate = restTemplate;
+        this.objectMapper = objectMapper;
+    }
```
Exception handling missing

DieS06 commented on demo/src/main/java/com/project/demo/rest/CountryController.java:
```diff
@@ -16,27 +15,32 @@ public class CountryController implements IController<Country, Integer>{
     private CountryService countryService;
```
Add service annotation.

mur-tz committed:
OpenAI Service Exception Handling

- Global Exception Handling additions
- Custom Exception Handling for OpenAI Service
- Created 3 new throwable exceptions

mur-tz reviewed (commented):
Already

mur-tz committed:
Comment again Integration Test
